include/wlr/backend.h

```diff
@@ -1,7 +1,30 @@
 #ifndef _WLR_BACKEND_H
 #define _WLR_BACKEND_H
 
-struct wlr_backend *wlr_backend_init();
-void wlr_backend_free(struct wlr_backend *backend);
+#include <wayland-server.h>
+
+struct wlr_backend_impl;
+struct wlr_backend_state;
+
+struct wlr_backend {
+	const struct wlr_backend_impl *impl;
+	struct wlr_backend_state *state;
+
+	struct {
+		struct wl_signal output_add;
+		struct wl_signal output_remove;
+		struct wl_signal output_frame;
+		struct wl_signal keyboard_add;
+		struct wl_signal keyboard_remove;
```

Comment:
Note that I don't expect the drm backend to handle these. There will eventually be a libinput backend that will emit these events, but not output events. Probably.